src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -1034,7 +1034,8 @@ private async Task<WorkUnitResult> ExecuteInstantiatedTask(ITaskExecutionHost ta
 
         private List<string> GetUndeclaredProjects(MSBuild msbuildTask)
         {
-            if (!_componentHost.BuildParameters.IsolateProjects)
+            ProjectIsolationMode isolateProjects = _componentHost.BuildParameters.ProjectIsolationMode;
```

Comment:
Same here; why isn't just if (isolateProjects == ProjectIsolationMode.False)?

Reply:
`GetUndeclaredProjects` is used to ensure that the `MSBuild` task doesn't execute on referenced projects that are undeclared when the isolation mode is set to `True`, as this would violate isolation constraints. If the isolation mode is either `Message` or `False`, then we're OK with such an execution because isolation violations are permitted.